test/AssetBasedIntentCurve.t.sol

```diff
@@ -4,132 +4,149 @@ pragma solidity ^0.8.13;
 /* solhint-disable func-name-mixedcase */
 
 import "./utils/TestEnvironment.sol";
+import "../src/test/AssetBasedIntentCurveLibHarness.sol";
 
 contract AssetBasedIntentCurveTest is TestEnvironment {
-    using AssetBasedIntentCurveLib for AssetBasedIntentCurve;
+    using AssetBasedIntentCurveLibHarness for AssetBasedIntentCurve;
 
-    AssetBasedIntentCurve internal _testConstantCurve =
-        _curveETH(AssetBasedIntentCurveBuilder.constantCurve(10), EvaluationType.ABSOLUTE);
-    AssetBasedIntentCurve internal _testConstantRelativeCurve =
-        _curveETH(AssetBasedIntentCurveBuilder.constantCurve(10), EvaluationType.RELATIVE);
-    AssetBasedIntentCurve internal _testLinearCurve =
-        _curveETH(AssetBasedIntentCurveBuilder.linearCurve(2, 10, 20, false), EvaluationType.ABSOLUTE);
-    AssetBasedIntentCurve internal _testLinearFlippedCurve =
-        _curveETH(AssetBasedIntentCurveBuilder.linearCurve(2, 10, 20, true), EvaluationType.ABSOLUTE);
-    AssetBasedIntentCurve internal _testExponentialCurve =
-        _curveETH(AssetBasedIntentCurveBuilder.exponentialCurve(2, 10, 2, 20, false), EvaluationType.ABSOLUTE);
-    AssetBasedIntentCurve internal _testExponentialFlippedCurve =
-        _curveETH(AssetBasedIntentCurveBuilder.exponentialCurve(2, 10, 2, 20, true), EvaluationType.ABSOLUTE);
+    function _constantCurve() private pure returns (AssetBasedIntentCurve memory) {
+        return _curveETH(AssetBasedIntentCurveBuilder.constantCurve(10), EvaluationType.ABSOLUTE);
```

Comment:
why did these have to be wrapped into functions?

Reply:
not a necessity by any means. looks cleaner and shadows the private functions `_intentForCase` and `_solutionForCase` that are used in scenarios

Reply:
Cool! That's fine